hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -120,16 +121,19 @@ object HoodieSparkSqlWriter {
 
     val fs = basePath.getFileSystem(sparkContext.hadoopConfiguration)
     tableExists = fs.exists(new Path(basePath, HoodieTableMetaClient.METAFOLDER_NAME))
-    var tableConfig = getHoodieTableConfig(sparkContext, path, hoodieTableConfigOpt)
-    val (parameters, hoodieConfig) = mergeParamsAndGetHoodieConfig(optParams, tableConfig, mode)
-    val originKeyGeneratorClassName = HoodieWriterUtils.getOriginKeyGenerator(parameters)
+    var tableConfig = if (mode == SaveMode.Overwrite || !tableExists) null else getHoodieTableConfig(sparkContext, path, hoodieTableConfigOpt)
```

Comment:
`tableExists` is handled in `getHoodieTableConfig` already. Also it's too early to check `SaveMode` here, which leads to confusion about why tableConfig being null while on storage it's non-null.